### DISCUSSION TITLE
oneof response bodies

### DISCUSSION THREAD

**Discussion by SebastianEngelmann** [juhaku/utoipa#902] [General]
Hi, I have a question regarding the response body. My API endpoint, which I created with actix-web, returns either struct a or struct b. Is there a way to model this with utoipa? I would need something like OneOf. Unfortunately I have not found an example in the docs. The code block is about lines 5 and 6.

```rust
#[utoipa::path(
    get,
    path = "/api/v1/app/{id}",
    responses(
        (status = 200, description = "struct a", body = StructA),
        (status = 200, description = "struct b", body = StructB),
        (status = 400, description = "No valid ID", body = String, example = json!(["no valid id"])),
        (status = 500, description = "Internal Server Error"),
        (status = NOT_FOUND, description= "ID was not found", example = json!(["no id found"]))
    ),
    params(
        ("id" = String, description = "id")
    )
)]
```

Is there a way I can tell swagger to return either struct a or struct b?

**Comment by juhaku** [comment]
@SebastianEngelmann There are 2 ways around this question. 
1. One either have different content type per response content as seen on the example in the docs: https://docs.rs/utoipa/latest/utoipa/attr.path.html#response-attributes _(Define multiple response return types)_. 
2. Create an enum out of your response struct.
    ```rust
    #[derive(ToSchema)]
    enum Response {
        A(StructA),
        B(StructB)
    }
    
    #[utoipa::path(
        get,
        path = "/api/v1/app/{id}",
        responses(
            (status = 200, description = "struct a or b", body = Response),
        ),
        params(
            ("id" = String, description = "id")
        )
    )]
    ```

**Reply by SebastianEngelmann** [reply]
Thank you so much!

**Reply by juhaku** [reply]
Welcome :slightly_smiling_face:
